### PR TITLE
xtest regression 1006: add tests on binary block properties

### PR DESCRIPTION
**Pull request by etienne-lms**
Added tests related to https://github.com/OP-TEE/optee_os/pull/3886.

### CODE REVIEW

**Comment by etienne-lms** [comment]
I expect IBART to fail as this P-R depends on https://github.com/OP-TEE/optee_os/pull/3886.

File: ta/os_test/os_test.c [hunk 1]
```diff
@@ -51,6 +51,90 @@ static TEE_Result check_returned_prop(
...
+
+	*bbuf = NULL;
+	*bblen = 0;
+	res = TEE_GetPropertyAsBinaryBlock(h, NULL, *bbuf, bblen);
+
+	if (res == TEE_SUCCESS && !*bblen)
+		return TEE_SUCCESS;
+
+	if (res != TEE_ERROR_SHORT_BUFFER) {
+		EMSG("TEE_GetPropertyAsBinaryBlock() szie query returned 0x%x\n",
```
Comment (jforissier) [inline ta/os_test/os_test.c:129]: size

File: ta/os_test/os_test.c [hunk 2]
```diff
@@ -51,6 +51,90 @@ static TEE_Result check_returned_prop(
...
+		     (unsigned int)res);
+		return TEE_ERROR_GENERIC;
```
Comment (jforissier) [inline ta/os_test/os_test.c:131]: return res?

File: ta/os_test/os_test.c [hunk 3]
```diff
@@ -51,6 +51,90 @@ static TEE_Result check_returned_prop(
...
+					size_t *bblen)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	*bbuf = NULL;
+	*bblen = 0;
+	res = TEE_GetPropertyAsBinaryBlock(h, NULL, *bbuf, bblen);
+
+	if (res == TEE_SUCCESS && !*bblen)
+		return TEE_SUCCESS;
```
Comment (jforissier) [inline ta/os_test/os_test.c:126]: This case would be an existing, zero-sized property, right? It is not currently tested, should it be added?

File: ta/os_test/os_test.c [hunk 4]
```diff
@@ -234,40 +320,47 @@ while (true) {
...
-				if (res != TEE_SUCCESS) {
-					EMSG(
-					"TEE_GetPropertyAsBinaryBlock(\"%s\") returned 0x%x\n",
-					nbuf, (unsigned int)res);
-					return res;
+			res = get_binblock_property(h, nbuf, &bbuf, &bblen);
+			if (res)
+				return res;
+
+			if (strcmp("myprop.binaryblock", nbuf) == 0) {
```
Comment (jforissier) [inline ta/os_test/os_test.c:335]: Minor: `!strcmp())`

File: ta/os_test/os_test.c [hunk 5]
```diff
@@ -51,6 +51,90 @@ static TEE_Result check_returned_prop(
...
+	case 4:
+		EMSG("\t%#"PRIx8" %#"PRIx8" %#"PRIx8" %#"PRIx8,
+		     bbuf[0], bbuf[1], bbuf[2], bbuf[3]);
+		break;
+	default:
+		TEE_Panic(0);
+	}
+
+	return TEE_ERROR_GENERIC;
+}
```
Comment (jforissier) [inline ta/os_test/os_test.c:112]: This is quite complex... how about:
```C
static TEE_Result check_binprop_ones(size_t size, uint8_t *bbuf, size_t bblen)  
{                                                                               
	uint8_t ones[4] = { 0xff, 0xff, 0xff, 0xff };                           

	if (size > 4 || bblen != size) {                                                    
		EMSG("Size error (size=%zu, bblen=%zu)", size, bblen);
		return TEE_ERROR_GENERIC;                                       
	}                                                                       
	if (strncmp(bbuf, ones, bblen)) {                                       
		EMSG("Unexpected content");                                     
		DHEXDUMP(bbuf, bblen);                                          
		return TEE_ERROR_GENERIC;                                       
	}                                                                       
	return TEE_SUCCESS;                                                     
}                                                                               
```

Comment (etienne-lms) [reply ta/os_test/os_test.c:112]: +1. I'll update the P-R accordingly.

File: ta/os_test/os_test.c (hunk 3 above)
Comment (etienne-lms) [reply ta/os_test/os_test.c:126]: This is (should be) tested by the 3 properties here: https://github.com/OP-TEE/optee_test/pull/432/files#diff-a518b40775b076c81be0dbf0be040d2bR41-R46.

File: ta/os_test/os_test.c (hunk 1 above)
Comment (etienne-lms) [reply ta/os_test/os_test.c:129]: thanks

File: ta/os_test/os_test.c (hunk 4 above)
Comment (etienne-lms) [reply ta/os_test/os_test.c:335]: ok

File: ta/os_test/os_test.c (hunk 3 above)
Comment (jforissier) [reply ta/os_test/os_test.c:126]: Duh, I reviewed only the first patch ;-) that is fine.

File: ta/os_test/os_test.c (hunk 2 above)
Comment (etienne-lms) [reply ta/os_test/os_test.c:131]: If `res == TEE_SUCCESS`, we must report another return code, hence i think `TEE_ERROR_GENERIC` would be fine. 
Maybe `return res ? res : TEE_ERROR_GENERIC;` ?

Comment (jforissier) [reply ta/os_test/os_test.c:131]: Ah you're right, no don't bother -- the original code is fine.

File: ta/os_test/os_test.c [hunk 6]
```diff
@@ -36,6 +36,14 @@ static TEE_Result check_returned_prop(
 		TEE_Result return_res, TEE_Result expected_res,
 		uint32_t return_len, uint32_t expected_len)
 {
+	/*
+	 * If expecting TEE_ERROR_SHORT_BUFFER but returned size matches
+	 * expected size then TEE_SUCCESS is a valid return code.
+	 */
+	if (expected_res == TEE_ERROR_SHORT_BUFFER &&
+	    return_res == TEE_SUCCESS && return_len <= expected_len)
+		return TEE_SUCCESS;
```
Comment (jforissier) [inline ta/os_test/os_test.c:45]: Why not change the caller instead? The caller knows if the size is big enough or not.

Comment (etienne-lms) [reply ta/os_test/os_test.c:45]: Ok, I agree i would be better.

**Comment by etienne-lms** [comment]
Updated.
IBART should succeed (OP-TEE/optee_os#3886 is merged).

**Pull request merged**